src/tests/ftest/deployment/ior_per_rank.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2018-2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+from avocado.core.exceptions import TestFail
+
+from ior_test_base import IorTestBase
+from ior_utils import IorCommand, IorMetrics
+from general_utils import percent_change
+
+
+class IorPerRank(IorTestBase):
+    # pylint: disable=attribute-defined-outside-init
+    # pylint: disable=too-few-public-methods
+    # pylint: disable=too-many-ancestors
+    """Test Class Description: Test class to run ior per rank basis and
+                               collect read/write data to identify bad
+                               nodes.
+    :avocado: recursive
+    """
+
+    def execute_ior_thread(self, rank):
+        """
+        Method to execute ior for different transfer sizes,
+        collect the performance numbers and match with the
+        expected values.
+        Args:
+            rank(str): Daos server rank on which ior needs to be run.
+        """
+
+        self.log.info("Running Test on rank: %s", rank)
+        # create the pool on specified rank.
+        self.create_pool([rank])
+        # execute ior on given rank for different transfer sizes and collect the results
+        for idx, _ in enumerate(self.transfer_sizes):
```

Comment:
Optional - for a direct approach
```suggestion
        for idx in range(len((self.transfer_sizes)):
```

Reply:
Thanks. I will prefer to stick with enum here. IMO it's better practice to use enum instead of `range(len(x))` (two function calls).